src/computational_graph/abstractgraph.jl

```diff
@@ -289,50 +289,45 @@ Base.:(==)(a::AbstractGraph, b::AbstractGraph) = Base.isequal(a, b)
 function isequiv(a::AbstractGraph, b::AbstractGraph, args...)
     typeof(a) != typeof(b) && return false
     # Check graph weights for approximate equality where applicable
-    if :weight ∉ args
-        (weight(a) ≈ weight(b)) == false && return false
+    if :weight ∉ args && !(weight(a) ≈ weight(b))
+        return false
     end
     # Check that all subgraphs are equivalent modulo `args`
-    length(subgraphs(a)) != length(subgraphs(b)) && return false
+    asubg_factors, bsubg_factors = subgraph_factors(a), subgraph_factors(b)
+    length(asubg_factors) != length(bsubg_factors) && return false
+
+    for field in fieldnames(typeof(a))
+        if field == :weight
+            continue  # skip graph weights if already accounted for
+        elseif field == :subgraphs
+            continue  # skip subgraphs if already accounted for
+        elseif field == :subgraph_factors
+            continue  # skip subgraph_factors if already accounted for
+        elseif field in args
+            continue  # skip fields in `args`
+        else
+            getproperty(a, field) != getproperty(b, field) && return false
+        end
+    end
 
-    # if :id ∉ args
-    #     pa = sortperm(subgraphs(a), by=x -> id(x))
-    #     pb = sortperm(subgraphs(b), by=x -> id(x))
-    #     subgraph_factors(a)[pa] != subgraph_factors(b)[pb] && return false
-    #     !all(isequiv.(subgraphs(a)[pa], subgraphs(b)[pb], args...)) && return false
-    # else
-    a_pairs = collect(zip(subgraphs(a), subgraph_factors(a)))
-    b_pairs = collect(zip(subgraphs(b), subgraph_factors(b)))
+    # Compare groups of subgraphs with the same factor
+    a_pairs = zip(subgraphs(a), asubg_factors)
+    b_pairs = collect(zip(subgraphs(b), bsubg_factors))
+    # b_pairs = zip(subgraphs(b), subgraph_factors(b))
+    # matched = falses(length(asubg_factors))
     for (suba, suba_factor) in a_pairs
```

Comment:
I don't see any need for `idx` and collected `b_pairs` here, we just want to look for the first non-match and return false asap if we do. We have also already tested that subg(a) and subg(b) are same-length at this stage. How about the following?

```
a_pairs = zip(subgraphs(a), asubg_factors)
b_pairs = zip(subgraphs(b), bsubg_factors)
for (suba, suba_factor) in a_pairs
    subequiv = false
    for (subb, subb_factor) in b_pairs
        return suba_factor == subb_factor && isequiv(suba, subb, args...)
    end
end
return true  # base case: suba and subb are empty
```